src/main/java/org/hiero/sketch/table/FullMembership.java

```diff
@@ -9,13 +9,16 @@
 public class FullMembership implements IMembershipSet {
     private final int rowCount;
 
-    public FullMembership(int rowCount) {
-        this.rowCount = rowCount;
+    public FullMembership(int rowCount) throws NegativeArraySizeException {
+            if (rowCount > 0)
+                this.rowCount = rowCount;
+            else
+                throw (new NegativeArraySizeException("Can't initialize FullMembership with negative rowCount"));
     }
 
     @Override
     public boolean isMember(int rowIndex) {
-        return rowIndex < this.rowCount;
+        return rowIndex < this.rowCount && rowIndex >= 0;
```

Comment:
I am torn about this. This may be in the innermost loop, and we don't want to check rowIndex >= 0 all the time, since the invariant will be established on initialization. I don't even know if we want the other test - perhaps this is inside a loop from 0 to size. Whatever decision we make we should document it in the API for IMembershipSet.